src/components/SecondaryPanes/Expressions.js

```diff
@@ -78,18 +79,31 @@ class Expressions extends PureComponent<Props, State> {
     deleteExpression(expression);
   }
 
-  inputKeyPress(e, expression) {
+  async inputKeyPress(e, expression) {
+    const target = e.target;
+    target.className = "input-expression";
+
     if (e.key !== "Enter") {
       return;
     }
 
-    const value = e.target.value;
+    const value = target.value;
+
     if (value == "") {
       return;
     }
 
+    const syntaxError = await hasSyntaxError(value);
+    this.setState({ error: syntaxError });
+
+    if (syntaxError) {
+      target.className = "input-expression-error";
```

Comment:
What if instead of editing `className` to `input-expression-error` we used the [classList](https://developer.mozilla.org/en-US/docs/Web/API/Element/classList) API (either the `.add()` or `.toggle()` methods) to add/remove an `error` CSS class so we have

```css
.input-expression
```

and

```css
.input-expression.error
```